src/screens/ChatScreen.tsx

```diff
@@ -0,0 +1,478 @@
+    !itemChat ||
+    isLoadingCurrentMember ||
+    !currentMember ||
+    isLoadingItemMemberships ||
+    !itemMemberships ||
+    !chatMembers
+  ) {
+    return null;
+  }
+  if (isErrorItemChat || isErrorCurrentMember || isErrorItemMemberships) {
+    return null;
+  }
+
+  const messages = convertToGiftedMessages(itemChat);
+
+  const postChatMessage = async (
+    newMessages: IMessage[] = [],
+    mentions?: string[],
+  ) => {
+    postMessage({
+      itemId,
+      body: newMessages[0].text,
+      mentions: mentions || [],
+    });
+  };
+
+  const patchChatMessage = async (editedMessages: IMessage[] = []) => {
+    if (messageSelected) {
+      patchMessage({
+        itemId,
+        messageId: messageSelected._id as string,
+        body: editedMessages[0].text,
+      });
+    }
+  };
+
+  const onSend = async (messages: IMessage[] = []) => {
+    const mentions: string[] = [];
+    const replacedMentionsMessage = [
+      {
+        ...messages[0],
+        text: replaceMentionValues(messages[0].text, ({ id, name }) => {
+          mentions.push(id);
+          return `\`<!@${id}>[${name}]\``;
+        }),
+      },
+    ];
+    if (messageSelected) {
+      await patchChatMessage(replacedMentionsMessage);
+      setMessageSelected(null);
+    } else {
+      await postChatMessage(replacedMentionsMessage, mentions);
+    }
+    setIsEditMessage(false);
+  };
+
+  const handleMessageOptions = (message: IMessage) => {
+    setMessageSelected(message);
+    handleOpenBottomSheetMessageOptionsModal();
+  };
+
+  const handleEditItemPress = (message: IMessage) => {
+    bottomSheetMessageOptionsModalRef.current?.close();
+    setIsEditMessage(true);
+    setInputMessage(replaceMessageWithMentions(message.text));
+  };
+
+  const handleDeleteItemPress = (messageId: UUID) => {
+    deleteMessage({
+      itemId,
+      messageId,
+    });
+    bottomSheetMessageOptionsModalRef.current?.close();
+    setMessageSelected(null);
+  };
+
+  const handleOnBackDropPressed = () => {
+    bottomSheetMessageOptionsModalRef.current?.close();
+    setIsEditMessage(false);
+    setMessageSelected(null);
+  };
+
+  const renderSuggestions: FC<any> = ({ keyword, onSuggestionPress }) => {
+    if (keyword == null) {
+      return null;
+    }
+    return (
+      <View style={styles.suggestionContainer}>
+        {chatMembers
+          .filter(({ name }) =>
+            name.toLowerCase().includes(keyword.toLowerCase()),
+          )
+          .map(({ id, name }) => (
+            <Pressable
+              key={id}
+              onPress={() => onSuggestionPress({ id, name })}
+              style={{ padding: 10 }}
+            >
+              <Text>{name}</Text>
+            </Pressable>
+          ))}
+      </View>
+    );
+  };
+
+  const renderComposer = (props: any) => {
+    const { text } = props;
+    return (
+      <View style={styles.composerContainer}>
+        {isEditMessage && (
+          <View
+            {...props}
```

Comment:
Avoid props spreading. It would be better to clearly define the props you need, this will also increase type-cheking safety